File: coding examples/Google Finance Stock Download Script/R/get_stock_price.R
rm(list=ls())

getprice<-function(q,i,p,f){
  s1<-'http://www.google.com/finance/getprices?q='
  s2='&i='
  s3='&p='
  s4='&f='
  url<-paste(s1, q, s2, i, s3, p, s4, f, sep = "")
  
  dat<-read.csv(url, header=F)
  df<-data.frame(dat[-c(1,2,3,4,5,6,7),])
  
  return(df)  
}

path="C:/xxx"
setwd(path)

symbol<-read.csv(file="symbols.csv", header=FALSE)
cp<-matrix(data=NA)

for(i in 1:nrow(symbol))
{
  q=symbol[i,1]
  cp[i]<-getprice(q,'300','15d','c')
}

t<-t(symbol)
names(cp)<-t

write.csv(cp,'price.csv')

File: machine learning models/Google Finance Stock Download Script/Python/get_stock_price.py
import urllib.request as urllib2
import csv
import numpy as np
import pandas as pd

filepath = 'c:/xxx/'

def getStockPrice(q, i = '300', p='15d', f='c'):
    # q: stock symbol
    # i: initerval
    # p: number of period
    # f: parameters
    url = "http://www.google.com/finance/getprices?q="+q+"&i="+i+"&p="+p+"&f="+f
    response = urllib2.urlopen(url)
    data = response.read().decode('utf-8').split('\n')
    a = data[7:len(data)]
    return a 

stocknames = pd.read_csv(filepath + "SymbolList.csv",header=None)

output = []
index = []

for i in range(0,len(stocknames)):
    ttl = stocknames[0][i]
    output1 = getStockPrice(q = ttl)
    output.append(output1) 
    index.append(ttl)

outprice = pd.DataFrame(output, index)
outprice = outprice.transpose()
outprice.to_csv(filepath + "results.csv", encoding='utf-8')

